### PR TITLE
Minor bugfixes that prevented first time startup

### PR DESCRIPTION
**Pull request by marpe**
This contains minor fixes I had to make to get things running.

Changes made:
* Prevent negative Width and Height values for `PreviewWindowItem` and `Location`
* Run `EnsureDirExists` before writing anything using `FileManager`
* Skip `doLayout` on disposed workspace
* Ignore calls to `SetIsShownInSwitchers` which throw `NotImplementedException`

More details in each commit message.

### CODE REVIEW

File: src/Whim.LayoutPreview/NonNegativeValueConverter.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+using Microsoft.UI.Xaml.Data;
+using System;
+
+namespace Whim.LayoutPreview;
+
+public class NonNegativeValueConverter : IValueConverter
```
Comment (dalyIsaac) [inline src/Whim.LayoutPreview/NonNegativeValueConverter.cs:6]: Could you make this `internal` for now, or add some cursory docs?

**Review by dalyIsaac — COMMENTED** [review]
Hi @marpe, thanks for contributing! Some of the `FloatingLayoutEngine` tests are failing, and there are a few tiny other things. Otherwise, it looks good to me, and I'll merge once the tests are fixed. Thanks again!

**Comment by marpe** [comment]
> Hi @marpe, thanks for contributing! Some of the `FloatingLayoutEngine` tests are failing, and there are a few tiny other things. Otherwise, it looks good to me, and I'll merge once the tests are fixed. Thanks again!

I got the tests passing by providing mocked values for monitor width/height, but you might want to double check those changes. I added the asserts when trying to figure out why the LayoutPreviewWindowItem width/height was negative, but maybe it doesn't make sense to keep em. I haven't looked closer at the different layout classes, maybe there's a use for negative widths/heights?

**Comment by dalyIsaac** [comment]
This is a bit of a can of worms. It's possible that layout engines could use negative widths or heights for internal calculations. However, I agree that the proxy layout engines should ensure that the widths and heights are non-negative where applicable.

I've written tests for `GapsLayoutEngine` to ensure that the widths and heights cannot be negative. I've added a similar test for `BarLayoutEngine`.

**Pull request merged**